conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt

```diff
@@ -0,0 +1,1317 @@
+Description: Test error in bitwise and expression where the operands are user defined type T
+             where the T is not a subtype of int.
+Labels: binary-bitwise-expr, DecimalNumber, int, module-type-defn
+
+type MyType int|decimal;
+
+function init() {
+    MyType a = 5;
+    MyType b = 433;
+    int c = 765;
+
+    int _ = a & b; // @error MyType is not a subtype of int
+    int _ = a & 12345; // @error MyType is not a subtype of int
+    int _ = b & c; // @error MyType is not a subtype of int
+    int _ = a & c; // @error MyType is not a subtype of int
+}
+
+Test-Case: output
+Description: Test if the type of either operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static
+             type of the result is int:UnsignedN where N is the smallest such K with Unsigned32.
+Labels: binary-bitwise-expr, DecimalNumber, int, int:Unsigned32
+
+function init() {
+    int a = 9876;
+    int:Unsigned32 b = 5454;
+    printResult(a & b); // @output 1028
+
+    int:Unsigned32 c = 9994444;
+    int:Unsigned32 d = 999999999;
+    printResult(c & d); // @output 9994444
+}
+
+function printResult(int:Unsigned32 result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test if the type of either operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static
+             type of the result is int:UnsignedN where N is the smallest such K with Unsigned16.
+Labels: binary-bitwise-expr, DecimalNumber, int, int:Unsigned16
+
+function init() {
+    int a = 33334567654;
+    int:Unsigned16 b = 213;
+    printResult(a & b); // @output 196
+
+    int:Unsigned16 c = 43343;
+    int:Unsigned16 d = 213;
+    printResult(c & d); // @output 69
+}
+
+function printResult(int:Unsigned16 result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test if the type of either operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static
+             type of the result is int:UnsignedN where N is the smallest such K with Unsigned8.
+Labels: binary-bitwise-expr, DecimalNumber, int, int:Unsigned8
+
+function init() {
+    int a = 987996;
+    int:Unsigned8 b = 54;
+    printResult(a & b); // @output 20
+
+    int:Unsigned8 c = 94;
+    int:Unsigned8 d = 99;
+    printResult(c & d); // @output 66
+}
+
+function printResult(int:Unsigned8 result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test if the type of either operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static
+             type of the result is int:UnsignedN where N is the smallest such K.
+Labels: binary-bitwise-expr, byte, DecimalNumber, int:Unsigned8, int:Unsigned16, int:Unsigned32
+
+function init() {
+    int:Unsigned32 b = 5454;
+    int:Unsigned16 d = 4344;
+    int:Unsigned16 e = b & d;
+    io:println(e); // @output 4168
+
+    int:Unsigned8 f = 44;
+    int:Unsigned8 g = d & f;
+    io:println(g); // @output 40
+
+    int:Unsigned8 h = f & g;
+    io:println(h); // @output 40
```

Comment:
both operands being int:Unsigned8 is covered in the test above. Shall we change this to cover Unsigned32 & Unsigned8 combination?